File: ts/bc1533.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let N: number | null = null;
let expectingSuspects = false; // Flag para saber se estamos esperando a linha dos suspeitos

rl.on('line', (line: string) => {
    if (!expectingSuspects) {
        N = parseInt(line.trim());

        if (N === 0) {
            rl.close();
            return;
        }

        expectingSuspects = true;
    } else {
        // Se estamos esperando suspeitos, esta é a linha deles
        const suspects = line.trim().split(' ').map(Number);

        // if (N === null || suspects.length !== N) {
        //     console.error("Erro");
        //     rl.close();
        //     return;
        // }

        let mostSuspect: number = -Infinity;
        let secondMostSuspect: number = -Infinity;

        for (const suspect of suspects) {
            if (suspect > mostSuspect) {
                secondMostSuspect = mostSuspect;
                mostSuspect = suspect;
            } else if (suspect > secondMostSuspect && suspect < mostSuspect) {
                secondMostSuspect = suspect;
            }
        }

        let iSecondMostSuspect: number = -1;
        for (let i = 0; i < N; i++) {
            if (suspects[i] === secondMostSuspect) {
                iSecondMostSuspect = i + 1;
                break;
            }
        }
        console.log(iSecondMostSuspect);

        expectingSuspects = false;
        N = null;
    }
});

rl.on('close', () => {
    // console.log("Programa encerrado."); // Opcional: mensagem ao fechar
    process.exit(0);
});

File: py/bc1533.py
import sys

def main():
    while True:
        N = int(sys.stdin.readline())
        if N == 0:
            break
        
        suspects = list(map(int, sys.stdin.readline().split()))

        most_suspect = 0
        second_most_suspect = 0

        for suspect in suspects:
            if suspect > most_suspect:
                second_most_suspect = most_suspect
                most_suspect = suspect
            elif suspect > second_most_suspect and suspect < most_suspect:
                second_most_suspect = suspect
        
        i_second_most_suspect = -1
        for i in range(N):
            if suspects[i] == second_most_suspect:
                i_second_most_suspect = i+1
                break
        
        print(i_second_most_suspect)

main()